internal/registry/clientsecretrequest/rest.go

```diff
@@ -6,27 +6,79 @@ package clientsecretrequest
 
 import (
 	"context"
+	"crypto/rand"
+	"encoding/hex"
 	"fmt"
+	"io"
+	"strings"
 
+	"golang.org/x/crypto/bcrypt"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	genericvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/api/validation/path"
 	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/rest"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/utils/trace"
 
 	clientsecretapi "go.pinniped.dev/generated/latest/apis/supervisor/clientsecret"
+	configv1alpha1clientset "go.pinniped.dev/generated/latest/client/supervisor/clientset/versioned/typed/config/v1alpha1"
+	"go.pinniped.dev/internal/oidcclientsecretstorage"
```

Comment:
Integration tests as a whole for this API:
- test all the variations of the `spec` (2 fields, all 4 combinations)
- ability to call the API multiple times to see the secret count update
- call the api multiple times, and prove you can authenticate with each of the secrets, each is valid
- calling revoke method should ensure you can auth with the latest secret only, previous secrets are revoked
- table test output for create
- multiple calls from separate go routines for the same oidc client.  some should succeed, some should fail, no data loss.  represent multiple users trying to interact with the same oidc client.
- if an oidc client is deleted, all secrets should be deleted. (create an oidc client, create a secret, delete the oidc client, then recreate it immediately, the old secret should not work.  the system should know those clients are not the same based on UID. even if the secret was not deleted the old secret, even if it still existed, should not be valid)

Reply:
- also, check and verify that the `kube secrets` are being deleted.

Reply:
- also check, when repeatedly adding new oidc client secrets, retain the original owner reference on the underlying kubernetes secret where these are stored.

Reply:
Other thoughts:
- test to preserve annotations, labels, etc?  
- we aren't retaining finalizers, is this ok